esphome/components/sml/sensor/__init__.py

```diff
@@ -0,0 +1,28 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import sensor
+from esphome.const import CONF_ID
+
+from .. import CONF_OBIS_CODE, CONF_SERVER_ID, CONF_SML_ID, Sml, obis_code, sml_ns
+
+SmlSensor = sml_ns.class_("SmlSensor", sensor.Sensor, cg.Component)
```

Comment:
This gives a better config error if they miss the main component.
```suggestion
from .. import CONF_OBIS_CODE, CONF_SERVER_ID, CONF_SML_ID, Sml, obis_code, sml_ns

DEPENDENCIES = ["sml"]

SmlSensor = sml_ns.class_("SmlSensor", sensor.Sensor, cg.Component)
```

But also since the component has no real config, you could auto load it instead:

```suggestion
from .. import CONF_OBIS_CODE, CONF_SERVER_ID, CONF_SML_ID, Sml, obis_code, sml_ns

AUTO_LOAD = ["sml"]

SmlSensor = sml_ns.class_("SmlSensor", sensor.Sensor, cg.Component)
```

Reply:
Added the AUTO_LOAD